File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

//Declare functions
void printPyramids(int n);
void printLine(int line_position, int total_lines);

int main(void)
{
    int pyramids_height;

    //Prompt for the size of de pyramid
    do
    {
        pyramids_height = get_int("Height: ");
    }
    while (pyramids_height <= 0 || pyramids_height > 8);

    printPyramids(pyramids_height);
}

void printPyramids(int max)
{
    //Recursively prints each line
    for (int i = 1; i <= max; i++)
    {
        printLine(i, max);
    }
}

void printLine(int line_position, int total_lines)
{

    int hashes_number = line_position;
    int spaces_number = total_lines - hashes_number;

    //Prints the initial space
    for (int i = 0; i < spaces_number; i++)
    {
        printf(" ");
    }

    //Prints the first triangle
    for (int i = 0; i < hashes_number; i++)
    {
        printf("#");
    }

    printf("  ");

    //Prints the second triangle
    for (int i = 0; i < hashes_number; i++)
    {
        printf("#");
    }

    printf("\n");
}

File: pset6/mario/more/mario.py
def pyramid(h):
    if h == 0:
        return

    # Draw pyramid of height h - 1
    pyramid(h - 1)

    # Print empty spaces
    for i in range(height - h):
        print(" ", end="")

    # Draw the first row of #
    for i in range(h):
        print("#", end="")

    # Print double space
    print("  ", end="")

    # Print the second row of spaces
    for i in range(h):
        print("#", end="")

    # Print a new line
    print()


while True:
    arg = input("Height: ")
    if arg.isnumeric():
        height = int(arg)
        if isinstance(height, int):
            if height > 0 and height <= 8:
                pyramid(height)
                break

